pkg/util/json/json.go

```diff
@@ -89,6 +89,10 @@ type JSON interface {
 	// Exists implements the `?` operator.
 	Exists(string) bool
 
+	// IterObjectKey returns an ObjectKeyIterator, and it returns nil if the obj
+	// is not an object.
+	IterObjectKey() *ObjectKeyIterator
```

Comment:
Perhaps we could just push the error down into this function, then it doesn't have to return a pointer (so the non-object cases would return an error with the "cannot be called on non-object" message)? I think it's fine if the error message for this case lives in the `json` package. It seems sensible to me to consider this an error case.